### PR TITLE
Manually place enemy zones

### LINKED ISSUE #33: Allow for manual placement of enemy zones
Some players wish to have a more custom experience. Allowing loose zone placement give a slightly less random experience but the zone type is still random. 

EXPECTED OUTCOME
New avenue is given to players for manual placement of enemy zones.

### PR DESCRIPTION
**Pull request by bolbies**
### Issue
Fixes #33

### Summary
First commit. I hope I'm doing this right!

### CODE REVIEW

**Comment by fritogotlayed** [comment]
@bolbies You're doing just fine. What issue number where you trying to resolve with this or was there another issue? If it was another issue I can create an issue and we can go from there. From this side it looks like you created the PR just fine :+1:

File: source/dialog/startup/startup.hpp [hunk 1]
```diff
@@ -156,16 +156,29 @@ class startup_dialog
...
+			text = "START-M"; //--- ToDo: Localize;
+			x = 24.5 * GUI_GRID_W + GUI_GRID_X;
+			y = 23 * GUI_GRID_H + GUI_GRID_Y;
+			w = 8 * GUI_GRID_W;
+			h = 1.5 * GUI_GRID_H;
+			sizeEx = 0.05;
+			colorBackground[] = {1, 0, 0,0.5};
+			tooltip = "Manually generate the zones, then start the campaign"; //--- ToDo: Localize;
+			action = "_nul = [] execvm ""dialog\startup\startup_start2.sqf""";
+		};
```
Comment (fritogotlayed) [inline source/dialog/startup/startup.hpp]: I know this is really nit-picky but can we please change the tab characters here to spaces? This appears to be the only block in this file that came across with tabs and not spaces. :-)

**Comment by fritogotlayed** [comment]
:eyes: Looking at this now.

File: source/dialog/startup/startup_start2.sqf [hunk 2]
```diff
@@ -0,0 +1,622 @@
...
+    {
+    commandpointsblu1 = 10000
+    };
+	
+	case index_amount_cp_1000:  
+    {
+    commandpointsblu1 = 1000
+    };
+	
+	case index_amount_cp_600:
```
Comment (fritogotlayed) [inline source/dialog/startup/startup_start2.sqf]: There's a tab here, need spaces :-)

File: source/dialog/startup/startup_start2.sqf [hunk 3]
```diff
@@ -0,0 +1,622 @@
...
+publicVariable "weather_type";
+publicVariable "blufor_ai_skill";
+publicVariable "opfor_ai_skill";
+publicVariable "enableChopperFastTravel";
+publicVariable "manually_chosen";
+
+	//	commandpointsblu1 = 9999999; 
+//hint format["Max radius: %1\nMin radius: %2\nZones number: %3\nCommand points: %4\nBLU AP: %5\nOPF AP: %6\nWeather type: %7\nBLU AI skill: %8\nOPF AI skill: %9",zones_max_radius,zones_min_radius,zones_number,commandpointsblu1,opfor_ap,blufor_ap,weather_type,blufor_ai_skill,opfor_ai_skill];
+diag_log format["----- DUWS CHOSEN SETTINGS --------- Max radius: %1-------Min radius: %2-------Zones number: %3-------Command points: %4-------BLU AP: %5-------OPF AP: %6-------Weather type: %7-------BLU AI skill: %8-------OPF AI skill: %9",zones_max_radius,zones_min_radius,zones_number,commandpointsblu1,opfor_ap,blufor_ap,weather_type,blufor_ai_skill,opfor_ai_skill];
+closeDialog 0;
```
Comment (fritogotlayed) [inline source/dialog/startup/startup_start2.sqf]: The formatting here is off. Can you please format the cases where the body inside of curly braces is kicked in? Example 1 or 2 is fine.
#### Example 1
```
switch(foo) do
{
    case bar:
    {
        blah command;
    }
};
```
#### Example 2:
```
switch(foo) do {
    case bar: {
        blah command;
    }
};
```

File: source/initHQ/BluHQinit.sqf [hunk 4]
```diff
@@ -89,9 +89,11 @@ if (!zones_manually_placed) then {
     // CALL ZONES GENERATION
     waitUntil {!isNil {getsize_script}};  // WAIT UNTIL THE MAPSIZE SCRIPT IS DONE
 
-    if (!zones_created) then {      // CHECK IF ZONES ARE PLACED, IF NOT EXECUTE locatorZonesV1.sqf
+    if (!zones_created && !manually_chosen) then {      // CHECK IF ZONES ARE PLACED, IF NOT EXECUTE locatorZonesV1.sqf
         _zones_create = [50, 0.2] execVM "initZones\locatorZonesV1.sqf";   // CHECK IF ZONES HAVE ALREADY BEEN PLACED
-    };
+	} else {
+		_zones_create = [50, 0.2] execVM "initZones\locatorZonesV2.sqf";
+	};
```
Comment (fritogotlayed) [inline source/initHQ/BluHQinit.sqf]: Tabs here as well.

File: source/initZones/locatorzonesV2.sqf [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+waitUntil {scriptDone _warcom_init};
+sleep 0.1;
+//savegame;
+enableSaving [ false, false ];
+sleep 1;
+
+[[format["MISSION INITIALIZATION COMPLETE!<br/>Campaign generated<br/>Welcome to %1, %2",_worldName,profilename]]] spawn BIS_fnc_typeText;  
+playMusic ["EventTrack01a_F_EPA", 0];
+
+if (debugmode) exitWith {};
```
Comment (fritogotlayed) [inline source/initZones/locatorzonesV2.sqf]: Tabs in this file as well.

**Comment by fritogotlayed** [comment]
@bolbies Done with my first pass review. I have to head out right now... dang wife...
Also, thanks a ton for the contribution! The purpose of the code review is not to attack you or your code but rather to make a higher quality mission for everyone :-). 

@naitp @phr0gz Feel free to review as well. I only spent about 10 minutes on the first pass. I'll review again later when I get back home.

Verdict - :no_good:

**Comment by bolbies** [comment]
I think I fixed the tabs. I'll make sure of that next time I post something. And the issue was from my own commit, not any of yours.

**Comment by fritogotlayed** [comment]
@bolbies I checked out your `manual-zones` branch and built the pbo to play local. When the game launched I manually placed my HQ. When I got to the zone screen I clicked the "START-M" button but nothing happens. Am I missing something here?

**Comment by bolbies** [comment]
I used the find/replace word tool to quickly turn the tabs into spaces and one of the variables at the bottom of startup_start2 got messed up. Sorry. Try it now.

**Comment by fritogotlayed** [comment]
@bolbies Ha. Not a problem. Have you tried an IDE like IntelliJ or SQF Editor? Both have an option to convert whitespace to spaces. It'll safely change tabs to spaces. I use both interchangeably.

File: source/dialog/startup/startup_start2.sqf [hunk 6]
```diff
@@ -0,0 +1,622 @@
...
+    {
+        blufor_ap = 110;
+    };
+	
+    case index_    blufor_ap_170:  
+    {
+        blufor_ap = 170;
+    };
+	
+    case index_    blufor_ap_200:
```
Comment (fritogotlayed) [inline source/dialog/startup/startup_start2.sqf]: I think these are messed up :-) `case index_    blufor_ap_...`

File: source/initZones/locatorzonesV2.sqf [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+            _missionPos = [ClickedPos, 0,50,15,0,0.1,0,[],[[0,0],[0,0]]] call BIS_fnc_findSafePos;
+
+        if (0 == _missionPos select 0 && 0 == _missionPos select 1) then {
+            clicked = false;
+            hint "Invalid position, the position must be flat and no objects must be near the position";
+        } else {
+            onMapSingleClick "";
+            hint "Valid pos, creating enemy zone";
+            _found_distance = true;
+            };
```
Comment (fritogotlayed) [inline source/initZones/locatorzonesV2.sqf]: the code in the if here should be kicked over as well. I had to read this twice to catch what was off in the alignment. The start of the if here should be even with line 37 :-)

File: source/initZones/locatorzonesV2.sqf [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+//hintSilent format["All zones found\nWelcome to %1\nHave fun !\nDUWS by kibot",_worldName];
+player globalChat format["All zones found. Welcome to %1, %2",_worldName,profileName];
+
+
+//hint format["MISSION INITIALIZATION COMPLETE!\nCampaign generated\nzones: %1\nmaximum radius: %2m\nminimum radius: %3m\nmax. distance from HQ: %4m\n\nIf you experience performance issues, restart the mission and try reducing the amount of zones/and or their radius",zones_number,zones_max_radius,zones_min_radius,zones_max_dist_from_hq];
+[[{player globalChat "MISSION INITIALIZATION COMPLETE!"}],"BIS_fnc_Spawn",true,false] call BIS_fnc_MP;
+[[{hint format["MISSION INITIALIZATION COMPLETE!\nCampaign generated\nzones: %1\nmaximum radius: %2m\nminimum radius: %3m\nmax. distance from HQ: %4m\n\nIf you experience performance issues, restart the mission and try reducing the amount of zones/and or their radius",zones_number,zones_max_radius,zones_min_radius,zones_max_dist_from_hq]}],"BIS_fnc_Spawn",true,false] call BIS_fnc_MP;
+[] spawn {
+sleep 9;
+[] execVM "misc\bottom_right_message.sqf";
```
Comment (fritogotlayed) [inline source/initZones/locatorzonesV2.sqf]: Code inside of this block needs to be indented as well.

**Comment by bolbies** [comment]
Guess who's never using the replace tool ever again? Fixes are up. Hopefully it works now!

**Comment by fritogotlayed** [comment]
Haha, it happens to all of us. I've been on the phone most of the evening and probably wont get to this until Wednesday. Thanks for making the edits! I strongly suggest using something like [Squint](http://www.armaholic.com/page.php?id=11817), [SQF Editor](http://www.armaholic.com/page.php?id=14714) or [IntelliJ](https://www.jetbrains.com/idea/) with the [SQF Plugin](http://www.armaholic.com/page.php?id=20194). All of the have an option to automatically convert tabs (`\t`) to spaces.

**Comment by bolbies** [comment]
Yeah I downloaded Squint so I'll see how that works out later when I start something else. I'll see if I can knock out a few of the posted issues next. This is fun :)

**Comment by fritogotlayed** [comment]
@bolbies All looked good to me from a code standpoint. Thanks for the addition. One area for improvement I'd like to see is that sometimes when I click nothing registers so I don't know if the script is analyzing to make sure the location I clicked is valid or if it's an invalid location. I'll create a new issue for this and we can address it in another PR. 

Also, I'm going to make some edits to get your PR into the format I'm planning to use going forward. This isn't a code edit; It's strictly just on this PR.

**Comment by fritogotlayed** [comment]
@bolbies You should join the @DUWS-R-Team/developers  team. We would love to have you!

**Pull request merged**